library/ziti.c

```diff
@@ -2137,28 +2137,38 @@ static void api_session_cb(ziti_api_session *api_sess, const ziti_error *err, vo
             goto done;
         }
 
-        struct tm exp;
-        ztx->id_creds.cert->get_expiration(ztx->id_creds.cert, &exp);
-        time_t now = time(0);
-        time_t exptime = mktime(&exp);
-
-        bool renew = exptime - now < ztx->opts.cert_extension_window * ONE_DAY;
-        if (!renew) {
+        if (!api_sess->is_cert_extendable) {
+            ZTX_LOG(DEBUG, "identity certificate is not renewable");
             goto done;
         }
 
-        if (!api_sess->is_cert_extendable) {
-            ZTX_LOG(WARN, "identity certificate is not renewable");
-            goto done;
+        struct tm exp;
+        if (api_sess->cert_extend_requested || api_sess->key_roll_requested) {
+            ZTX_LOG(INFO, "controller requested certificate renewal (%s key roll)",
+                    api_sess->key_roll_requested ? "with" : "without");
+        } else {
+            ztx->id_creds.cert->get_expiration(ztx->id_creds.cert, &exp);
+            time_t now = time(0);
+            time_t exptime = mktime(&exp);
+
+            bool renew = exptime - now < ztx->opts.cert_extension_window * ONE_DAY;
```

Comment:
i know it used to be one day but one day really doesn't feel like enough imo unless when it comes back online it's allowed to immediately extend

Reply:
it isn't one day?